internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -323,6 +339,7 @@ func (r *NumaflowControllerRolloutReconciler) reconcile(
 		return ctrl.Result{}, fmt.Errorf("sync operation is not successful")
 	}
 
+	r.customMetrics.ReconciliationDuration.WithLabelValues(ControllerNumaflowControllerRollout, "create").Observe(time.Since(syncStartTime).Seconds())
```

Comment:
I don't think this one is necessary "create", right? It could be auto-healing. It could be a no-op.

Reply:
hmm...in that case I believe need to add an if condition, something like
if !deploymentExists {
...
}